data/service/data-service.js

```diff
@@ -272,16 +272,21 @@ exports.DataService = Montage.specialize(/** @lends DataService.prototype */ {
     _addChildService: {
         value: function (child, types) {
             var children, type, i, n, nIfEmpty = 1;
-            types = types || child.model && child.model.objectDescriptors || child.types;
+            types = types || child.model && child.model.objectDescriptors || child.types,
+            isNotParentService = (child._parentService !== this);
```

Comment:
Positive variables are generally clearer, i.e. `isParentService` and `!isParentService`.